core/pkg/resultshandling/reporter/v2/mockreporter_test.go

```diff
@@ -1,10 +1,37 @@
 package reporter
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
 	"testing"
+
+	"github.com/armosec/armoapi-go/armotypes"
+	"github.com/kubescape/k8s-interface/workloadinterface"
+	"github.com/kubescape/kubescape/v2/core/cautils"
+	"github.com/kubescape/kubescape/v2/core/pkg/resultshandling/reporter"
+	"github.com/kubescape/opa-utils/reporthandling"
+	"github.com/kubescape/opa-utils/reporthandling/apis"
+	"github.com/kubescape/opa-utils/reporthandling/attacktrack/v1alpha1"
+	"github.com/kubescape/opa-utils/reporthandling/results/v1/prioritization"
+	"github.com/kubescape/opa-utils/reporthandling/results/v1/resourcesresults"
+	reporthandlingv2 "github.com/kubescape/opa-utils/reporthandling/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestReportMock_GetURL(t *testing.T) {
+func TestReportMockGetURL(t *testing.T) {
+	t.Parallel()
```

Comment:
question: we have so many tests that we need to specify this everywhere? I don't remember we use this so much in kubernetes/kubernetes

Reply:
The "t.Parallel()" clause is not because we have a lot of tests. The test runtime just spawns go routines according to the limits of the environment (e.g. go test -parallel n). I am inserting this almost everywhere. But I may refrain from doing it.

Reply:
I was just curious, thx!

Reply:
@matthyx in some situations, there is a catch when too many tests are parallelized, it's when your tested program overcomes some system limit (e.g. open file descriptors) when go routines are too lazy. In such cases, a massive number of test routines may uncover issues and we have to refrain...

Reply:
yeah we had that sometimes in k/k... but we didn't need to run test cases in parallel for that